server/src/main/java/org/elasticsearch/cluster/metadata/Template.java

```diff
@@ -180,7 +206,15 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeMap(this.aliases, StreamOutput::writeString, (stream, aliasMetadata) -> aliasMetadata.writeTo(stream));
         }
         if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0) && DataLifecycle.isEnabled()) {
-            out.writeOptionalWriteable(lifecycle);
+            if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_007)) {
+                boolean isExplicitNull = NO_LIFECYCLE.equals(lifecycle);
```

Comment:
We shouldn't use `equals` here but `==`

Should we also add a unit test that fails if `equals` is used here?